src/objects-table/ObjectsTable.js

```diff
@@ -248,15 +268,63 @@ class ObjectsTable extends React.Component {
         return this.actions.isContextActionAllowed(this.props.d2, ...args);
     };
 
+    getSelectionMessages = () => {
+        const { allObjects, dataRows, selection, pager } = this.state;
+        const messages = [];
+
+        const selectedAllInCurrentPage =
+            !_.isEmpty(dataRows) && dataRows.every(row => selection.has(row.id));
+        const selectedAllInAllPages = selection.size === pager.total;
+        const selectInAllPagesImplemented = allObjects.size === pager.total;
+
+        if (selectedAllInCurrentPage && !selectedAllInAllPages && selectInAllPagesImplemented) {
+            messages.push({
+                message: i18n.t("All {{count}} items on this page are selected.", {
+                    count: dataRows.length,
+                }),
+                link: i18n.t("Select all {{count}} items in all pages", { count: pager.total }),
+                action: this.selectAllPages,
+            });
+        }
+
+        const selectionInOtherPages = _.difference(
+            Array.from(selection),
+            dataRows.map(dr => dr.id)
+        );
+
+        if (selectedAllInAllPages) {
+            messages.push({
+                message: i18n.t("There are {{count}} items selected in all pages.", {
```

Comment:
There is an edge case with a un-intuitive. message  Let's imagine I have all items (2010) selected, then I see: `There are 2010 items selected in all pages`. I unselect one, now I see `There are 1990 items selected on other pages`. It's correct, but maybe we could show also the total? `There are 2009 items selected (1990 in other pages)`. Not sure if this makes sense, just as an idea.

Reply:
Makes sense